lnd.go

```diff
@@ -1627,106 +1564,141 @@ func initializeDatabases(ctx context.Context,
-		)
-		switch {
-		case err == channeldb.ErrDryRunMigrationOK:
-			return nil, nil, err
+	if databaseBackends.Replicated {
+		ltndLog.Infof("Database replication is available! Creating " +
+			"graph and channel state DB instances")
+	} else {
+		ltndLog.Infof("Creating local graph and channel state DB " +
+			"instances")
+	}
+
+	// Otherwise, we'll open two instances, one for the state we only need
+	// locally, and the other for things we want to ensure are replicated.
+	dbs.graphDB, err = channeldb.CreateWithBackend(
+		databaseBackends.GraphDB,
+		channeldb.OptionSetRejectCacheSize(cfg.Caches.RejectCacheSize),
+		channeldb.OptionSetChannelCacheSize(cfg.Caches.ChannelCacheSize),
+		channeldb.OptionSetBatchCommitInterval(cfg.DB.BatchCommitInterval),
+		channeldb.OptionDryRunMigration(cfg.DryRunMigration),
+	)
+	switch {
+	// As we want to allow both versions to get thru the dry run migration,
+	// we'll only exit the second time here once the remote instance has had
+	// a time to migrate as well.
+	case err == channeldb.ErrDryRunMigrationOK:
+		ltndLog.Infof("Graph DB dry run migration successful")
 
-		case err != nil:
-			err := fmt.Errorf("unable to open local channeldb: %v", err)
-			ltndLog.Error(err)
-			return nil, nil, err
-		}
+	case err != nil:
+		cleanUp()
 
-		closeFuncs = append(closeFuncs, func() {
-			dbs.graphDB.Close()
-		})
+		err := fmt.Errorf("unable to open graph DB: %v", err)
+		ltndLog.Error(err)
+		return nil, nil, err
+	}
 
-		dbs.chanStateDB = dbs.graphDB
-	} else {
-		ltndLog.Infof("Database replication is available! Creating " +
-			"local and remote channeldb instances")
-
-		// Otherwise, we'll open two instances, one for the state we
-		// only need locally, and the other for things we want to
-		// ensure are replicated.
-		dbs.graphDB, err = channeldb.CreateWithBackend(
-			databaseBackends.GraphDB,
-			channeldb.OptionSetRejectCacheSize(cfg.Caches.RejectCacheSize),
-			channeldb.OptionSetChannelCacheSize(cfg.Caches.ChannelCacheSize),
-			channeldb.OptionSetBatchCommitInterval(cfg.DB.BatchCommitInterval),
-			channeldb.OptionDryRunMigration(cfg.DryRunMigration),
-		)
-		switch {
-		// As we want to allow both versions to get thru the dry run
-		// migration, we'll only exit the second time here once the
-		// remote instance has had a time to migrate as well.
-		case err == channeldb.ErrDryRunMigrationOK:
-			ltndLog.Infof("Local DB dry run migration successful")
-
-		case err != nil:
-			err := fmt.Errorf("unable to open local channeldb: %v", err)
-			ltndLog.Error(err)
-			return nil, nil, err
-		}
+	// Success, let's overwrite the close function with the decorated one.
+	closeFuncs["graph"] = dbs.graphDB.Close
 
-		closeFuncs = append(closeFuncs, func() {
-			dbs.graphDB.Close()
-		})
+	ltndLog.Infof("Opening channel state database instance...")
 
-		ltndLog.Infof("Opening replicated database instance...")
+	dbs.chanStateDB, err = channeldb.CreateWithBackend(
```

Comment:
Isn't this applying migrations twice to the same underlying db now?

Reply:
Hmm, you're right. I think we can't split it here yet until we do the actual separation of the two databases on the code level. Also added a TODO.